src/Diagnostics/FxCop/CSharp/Design/CodeFixes/CA1052CSharpCodeFixProvider.cs

```diff
@@ -60,4 +78,24 @@ private class MyCodeAction : DocumentChangeAction
             }
         }
     }
+
+    internal static class CA1052CSharpCodeFixProviderExtensions
+    {
+        internal static bool IsDefaultConstructor(this MemberDeclarationSyntax member)
+        {
+            if (member.Kind() != SyntaxKind.ConstructorDeclaration)
+            {
+                return false;
+            }
+
+            var constructor = (ConstructorDeclarationSyntax)member;
+            if (constructor.Modifiers.Any(m => m.Kind() == SyntaxKind.StaticKeyword))
+            {
+                return false;
+            }
+
+            return constructor.ParameterList.Parameters.Count == 0;
```

Comment:
No, there is always a parameter list syntax node (the parentheses must always be there), and there is always a Parameters list inside it (even if the list is empty).